code/modules/hydroponics/spreading/spreading_response.dm

```diff
@@ -154,23 +162,33 @@
 /obj/effect/plantsegment/proc/entangle_mob(var/mob/living/victim)
 	if(!victim || victim.locked_to || !seed || seed.spread != 2 || is_locking(/datum/locking_category/plantsegment)) //How much of this is actually necessary, I wonder
 		return
-
+	if(is_protected(victim))
+		return
 	lock_atom(victim, /datum/locking_category/plantsegment)
 	if(victim.stat != DEAD)
 		to_chat(victim, "<span class='danger'>The vines [pick("wind", "tangle", "tighten")] around you!</span>")
 
 /obj/effect/plantsegment/proc/grab_mob(var/mob/living/victim)
 	if(!victim || victim.locked_to || !Adjacent(victim)|| !seed || seed.spread != 2 || is_locking(/datum/locking_category/plantsegment))
 		return
-
 	var/can_grab = 1
 	if(istype(victim, /mob/living/carbon/human))
 		var/mob/living/carbon/human/H = victim
 		if(istype(H.shoes, /obj/item/clothing/shoes/magboots) && (H.shoes.clothing_flags & NOSLIP))
 			can_grab = 0
+		if(is_protected(victim))
+			can_grab = 0
 	if(can_grab)
 		src.visible_message("<span class='danger'>Tendrils lash out from \the [src] and drag \the [victim] in!</span>")
 		victim.forceMove(src.loc)
 		lock_atom(victim, /datum/locking_category/plantsegment)
 
+/obj/effect/plantsegment/proc/is_protected(var/mob/living/carbon/human/H)
```

Comment:
why is this a proc on `/obj/effect/plantsegment`?

Reply:
There's no good reason why. I've moved the proc to /mob/living